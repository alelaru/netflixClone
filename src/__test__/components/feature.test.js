import { queryByTestId, render } from "@testing-library/react"
import Feature from "../../components/feature"


describe("<Feature/>", () => {

    it("Renders the <Feature/> with populated data", () => {

        const {container, getByText } = render(

            <Feature>
              <Feature.Title>Unlimited films, tv, programs and more</Feature.Title>
              <Feature.SubTitle>Watch anywhere and cancel anytime</Feature.SubTitle>
            </Feature>
        )

        expect(getByText("Unlimited films, tv, programs and more")).toBeTruthy();
        expect(getByText("Watch anywhere and cancel anytime")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    })

    it("Renders the <Feature/> with just the title", () => {

        const {container, getByText, queryByText } = render(

            <Feature>
              <Feature.Title>Unlimited films, tv, programs and more</Feature.Title>
            </Feature>
        )

        expect(getByText("Unlimited films, tv, programs and more")).toBeTruthy();
        expect(queryByText("Watch anywhere and cancel anytime")).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();

    })

})