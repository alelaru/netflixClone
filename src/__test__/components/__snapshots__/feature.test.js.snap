// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> Renders the <Feature/> with just the title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cXQSCO"
>
  <h1
    class="feature__Title-sc-6swrp6-1 emDtDo"
  >
    Unlimited films, tv, programs and more
  </h1>
</div>
`;

exports[`<Feature/> Renders the <Feature/> with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cXQSCO"
>
  <h1
    class="feature__Title-sc-6swrp6-1 emDtDo"
  >
    Unlimited films, tv, programs and more
  </h1>
  <h2
    class="feature__Subtitle-sc-6swrp6-2 ihOZwE"
  >
    Watch anywhere and cancel anytime
  </h2>
</div>
`;
